tests/test_util.py

```diff
@@ -225,3 +227,17 @@ def test_set_block():
         full_arr = np.random.random((10, 12))
         block_arr = np.zeros((2, 2))
         mod_arr = set_block(full_arr, block_arr)
+
+
+def test_filename_from_url():
+    letters = np.array(list(string.printable))
+    fake = Faker()
```

Comment:
Good idea with Faker!
Great job, but please add also 2-3 string urls to the test function - we also have that as "documentation" to see what and how the function works